### PR TITLE
Week 3, Walkthrough #6: Requesting Number of Comments

### PR DESCRIPTION
**Pull request by kashiish**
These commits add the ability to request a specific number of comments from the Datastore. In DataServlet.java, the number of comments fetched from Datastore is limited by the max-comments parameter. The parameter is checked for valid user input. In index.html, I created a new select form for the user to select either 5, 10, or 25 comments. In script.js, I have updated the fetch query string to include the max-comments parameter.

### CODE REVIEW

**Comment by kashiish** [comment]
Not sure what I did, but some of the commits are appearing twice. Sorry about that!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -92,12 +126,12 @@ private String convertListToJson(ArrayList<Comment> comments) {
    * @return the request parameter, or the default value if the parameter
    *         was not specified by the client
    */
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
-    String value = request.getParameter(name);
-    if (value == null) {
-      return defaultValue;
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (dieseldesai) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:129]: I'd suggest using Optionals here. This method can return an Optional<String> and then the caller can use the method Optional.orElse("5") to use a default value

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Comment (kashiish) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I tried implementing with Optionals. Let me know what you think.

Comment (dieseldesai) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nice! LGTM

**Pull request merged**